File: game-frontend/projects/game-frontend/src/app/modules/ranking/types/ranking-entry.ts
/**
 * Represents a ranking entry
 *
 * @author Kevin Guanche Darias <kevin@kevinguanchedarias.com>
 * @since 0.7.0
 * @export
 * @interface RankingEntry
 */
export interface RankingEntry {
    position: number;
    points: number;
    userId: number;
    username: string;
    allianceId: number;
    allianceName: string;
}

File: business/src/main/java/com/kevinguanchedarias/owgejava/pojo/RankingEntry.java
/**
 *
 */
package com.kevinguanchedarias.owgejava.pojo;

import com.kevinguanchedarias.owgejava.dto.CommonDtoWithImageStore;
import com.kevinguanchedarias.owgejava.entity.Faction;

/**
 * Represents a ranking entry
 *
 * @author Kevin Guanche Darias <kevin@kevinguanchedarias.com>
 * @since 0.7.0
 */
public record RankingEntry(
        Number position,
        Number points,
        Number userId,
        String username,
        Number allianceId,
        String allianceName,
        CommonDtoWithImageStore<Integer, Faction> faction
) {
}
